tokio-macros/src/lib.rs

```diff
@@ -74,16 +74,18 @@ pub fn main(args: TokenStream, item: TokenStream) -> TokenStream {
     let mut runtime = RuntimeType::Multi;
 
     for arg in args {
-        if let syn::NestedMeta::Meta(syn::Meta::Path(ident)) = arg {
-            let seg = ident.segments.first().expect("Must have specified ident");
-            match seg.ident.to_string().to_lowercase().as_str() {
+        if let syn::NestedMeta::Meta(syn::Meta::Path(path)) = arg {
+            let ident = path.get_ident();
+            if ident.is_none() {
+                let msg = "Must have specified ident";
```

Comment:
Before: 

*(passed)*

After:

```text
error: Must have specified ident
  --> tokio/examples/chat.rs:41:15
   |
41 | #[tokio::main(multi_thread::P)]
   |               ^^^^^^^^^^^^^^^
```

Reply:
The original code only checks that the first element in the path is `multi_thread` or `single_thread`.